src/game/common/system/mempoolobj.h

```diff
@@ -122,11 +122,9 @@ protected: \
  */
 inline void MemoryPoolObject::Delete_Instance()
 {
-    if (this != nullptr) {
-        MemoryPool *pool = Get_Object_Pool();
-        this->~MemoryPoolObject();
-        pool->Free_Block(this);
-    }
+    MemoryPool *pool = Get_Object_Pool();
```

Comment:
Not sure about this one. Although not standard i think we can have nullptr this.